### ISSUE TITLE
Asset helper & asset related amends

### ISSUE THREAD

**Issue by ErisDS** [TryGhost/Ghost#8221]
Refs #7491, #7488 (Theme & Config Audit)

After #8126 is finished & merged, we will be able to reduce the amount of logic in the asset helper & the related `data/meta/asset_url.js` utility.

These no longer need to handle any assets for the admin panel - dealing only with server & shared assets. 

#### Part 1: Get rid of the `ghost="true"` concept

We no longer need to serve any admin/client assets from templates inside of Ghost. The concept of ghost=true can therefore go away entirely. This is relatively simple.

#### Part 2: Error handling hbs engine

If the error template still requires the asset helper, which it may, we need to ensure that it is always available to the handlebars engine in error-handling.js. 

#### Part 3: Colocate assets

Iwould be beneficial to colocate the assets that need to be served for the frontend. If Apps need to supply assets (which they probably do) there should be a specific file structure for them.

#### Part 4: Get rid of `minifyInProduction` / `minifyAsset`

We have this attribute, which sounds like we have a build pipeline to do some sort of minification. Actually, it changes the filename from `x.y` to `x.min.y`, so that we can serve the pre-minified version of an asset in production mode. It's only used in internal stuff, not documented for the world,  but it is available and therefore at the risk of being used! 

It is also paired with a config value of `minifyAssets` which sounds even more like we have a build pipeline and is currently used instead of checking the env, which makes the attribute `minifyInProduction` not make sense.

I *think* this concept needs to either go away, or be replaced with a new concept. Maybe `useDotMinAssets` as both the attribute and config value names?

There are 2 things that make me think this concept can't go away entirely:

1. We have this usage [in ghost_head](https://github.com/TryGhost/Ghost/blob/master/core/server/helpers/ghost_head.js#L70) for serving ghost-url.js vs ghost-url.min.js.
2. And we have this issue https://github.com/TryGhost/Ghost/issues/7314 with AMP CSS.

Also in #8126 we may want to have minified and unminified versions of the CSS.

#### Part 5: Reconsider asset hashing

There is a TODO in the new themes/active.js that talks about moving the concept of asset hashing out of config. 

There are 2 possible solutions
- quick: move assetHash to be a property of the active theme
- better: use a proper asset hashing lib to get files with names like x.3423abc23.js like we have on the client side

### Bottom line: do we need to have a (better) build pipeline?

For production, we have a number of assets around - images, js and css, that need to be minified/uglified to serve for production. We currently do this on `grunt prod` for some assets, including ghost-url.js. For other assets we store both versions in the repo. 

Perhaps it makes sense to provide for minifying both JS and CSS, and allow files that need to be minified to be supplied via config, rather than in the Gruntfile?

This would work for internal apps + Ghost itself.

**Comment by ErisDS** [comment]
@AileenCGN there seems to be 2 instances of ghosticon.jpg still being used in the serverside codebase - one in meta/index.js and one in slack/index.js.

**Comment by AileenCGN** [comment]
I've looked into the possible usage of an `urlFor('asset')` util and here's my conclusion:

Right now, we're calling the `getAssetUrl` fn from our `{{asset}}` and `{{ghost_head}}` helper in `server/helpers`. `getAssetUrl` lives in `server/data/meta`, *but* is never used in meta data itself. IMO it would be better of as a util, specifically a URL util, which we could call with `urlFor('asset')` and make it therefore more consistent within our usage of utils in the codebase.
